lib/rack/attack/throttle.rb

```diff
@@ -23,34 +23,48 @@ def cache
 
       def matched_by?(request)
         discriminator = block.call(request)
+
         return false unless discriminator
 
-        current_period = period.respond_to?(:call) ? period.call(request) : period
-        current_limit  = limit.respond_to?(:call) ? limit.call(request) : limit
-        key            = "#{name}:#{discriminator}"
-        count          = cache.count(key, current_period)
-        epoch_time     = cache.last_epoch_time
+        current_period  = current_period(request)
+        current_limit   = current_limit(request)
```

Comment:
Can we please use different names for local variable and instance methods? I think it helps avoid possible confusion for the developer :-)

Some ideas:
- `period_for(request)` and `limit_for(request)` or
-  `request_period(request)` and `request_limit(request)`

Reply:
Agree, could be misleading. Fixed!